include/astarte_device.h

```diff
@@ -29,8 +30,7 @@ typedef struct
     astarte_device_handle_t device;
     const char *interface_name;
     const char *path;
-    const void *bson_value;
-    int bson_value_type;
+    astarte_bson_element_t bson_element;
```

Comment:
I'm not sure about this change, it introduces a breaking change when receiving the ` astarte_device_data_event_t` . It may be better to add in another milestone and keeping track by an issue.